java/src/lib.rs

```diff
@@ -17,20 +17,23 @@ pub use ffi_test::*;
 
 // TODO: Consider caching method IDs here in a static variable (might need RwLock to mutate)
 fn redis_value_to_java<'local>(env: &mut JNIEnv<'local>, val: Value) -> JObject<'local> {
+    //dbg!(val.clone());
     match val {
         Value::Nil => JObject::null(),
         Value::SimpleString(str) => JObject::from(env.new_string(str).unwrap()),
         Value::Okay => JObject::from(env.new_string("OK").unwrap()),
         Value::Int(num) => env
             .new_object("java/lang/Long", "(J)V", &[num.into()])
             .unwrap(),
-        Value::BulkString(data) => match std::str::from_utf8(data.as_ref()) {
-            Ok(val) => JObject::from(env.new_string(val).unwrap()),
-            Err(_err) => {
-                let _ = env.throw("Error decoding Unicode data");
-                JObject::null()
-            }
-        },
+        Value::BulkString(data) => {
+            // TODO handle unwraps
+            let array = env.new_byte_array(data.len() as i32).unwrap();
+            // TODO convert &[u8] to &[i8] without `unsafe`
+            let i8slice = unsafe { &*(data.as_slice() as *const [u8] as *const [i8]) };
```

Comment:
why are we taking the `unsafe` path? we should avoid this at all cost

Reply:
Already fixed